internal/replacements/replacements.go

```diff
@@ -257,6 +265,51 @@ func resourcesNeedsReplacement(desired []corev1.Container, current []corev1.Cont
 	return desiredCPURequests.Cmp(*currentCPURequests) == 1 || desiredMemoryRequests.Cmp(*currentMemoryRequests) == 1
 }
 
+// fileSecurityContextChanged checks for changes in the effective security context by checking that there are no changes
+// to the following SecurityContext (or PodSecurityContext) fields:
+// RunAsGroup, RunAsUser, FSGroup, or FSGroupChangePolicy
+// See https://github.com/FoundationDB/fdb-kubernetes-operator/issues/208 for motivation
+// only makes sense if both pods have containers with matching names
+func fileSecurityContextChanged(desired, current *corev1.Pod) bool {
```

Comment:
We should add some logging here, similar to what we do in the other replacement checks to let the user know why the process group was replaced.